scripts/mbedtls_dev/macro_collector.py

```diff
@@ -105,6 +105,16 @@ def __init__(self) -> None:
             'tag_length': [],
             'min_tag_length': [],
         } #type: Dict[str, List[str]]
+        self.include_intermediate = False
```

Comment:
I've completely reworded the commit message which was inaccurate (`psa_generate_tests` is still using `InputsForTests`, what changed is the base class of `InputsForTests`).

I agree that this commit is messy. I don't think I split it right. But each commit was somewhat tested and it would be a nontrivial effort to go back and re-split into meaningful, tested commits. I don't think it's worth it at this point. `macro_collector.py` needs further work to clarify the classes, what they can do and what they're intended for. Completing this work is a long-term background effort that goes beyond the scope of the current pull request.